{"expand":"renderedFields,names,schema,operations,editmeta,changelog,versionedRepresentations","id":"12458864","self":"https://issues.apache.org/jira/rest/api/2/issue/12458864","key":"OPENJPA-1565","fields":{"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12314857","id":"12314857","description":"2.0.0 Beta 3","name":"2.0.0-beta3","archived":false,"released":true,"releaseDate":"2010-03-29"}],"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12312322":null,"customfield_12312323":null,"customfield_12312320":null,"customfield_12310420":"161828","customfield_12312321":null,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312720":null,"customfield_12312325":null,"lastViewed":null,"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12313422":"false","customfield_12312331":null,"customfield_12312332":null,"aggregatetimeoriginalestimate":null,"timeestimate":null,"customfield_12312330":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313624","id":"12313624","description":"OpenJPA 2.0.0 Milestone 1","name":"2.0.0-M1","archived":false,"released":true,"releaseDate":"2009-01-29"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313483","id":"12313483","description":"Early Access 2","name":"2.0.0-M2","archived":false,"released":true,"releaseDate":"2009-06-03"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12314148","id":"12314148","description":"Early Access 3","name":"2.0.0-M3","archived":false,"released":true,"releaseDate":"2009-10-12"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12314149","id":"12314149","description":"2.0.0 Beta","name":"2.0.0-beta","archived":false,"released":true,"releaseDate":"2010-01-28"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12314802","id":"12314802","description":"2.0.0 Beta 2","name":"2.0.0-beta2","archived":false,"released":true,"releaseDate":"2010-02-27"}],"customfield_12311120":null,"customfield_12313826":null,"customfield_12312339":null,"issuelinks":[{"id":"12330907","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12330907","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"outwardIssue":{"id":"12459907","key":"OPENJPA-1594","self":"https://issues.apache.org/jira/rest/api/2/issue/12459907","fields":{"summary":"Tests not handling new QueryTimeOut and LockTimeOut exceptions correctly","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/7","id":"7","description":"The sub-task of the issue","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21146&avatarType=issuetype","name":"Sub-task","subtask":true,"avatarId":21146}}}},{"id":"12331246","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12331246","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"outwardIssue":{"id":"12460975","key":"OPENJPA-1609","self":"https://issues.apache.org/jira/rest/api/2/issue/12460975","fields":{"summary":"PessimisticLockException instead of LockTimeoutException thrown on DB2V9 for ZOS","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133}}}},{"id":"12331245","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12331245","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"outwardIssue":{"id":"12461400","key":"OPENJPA-1614","self":"https://issues.apache.org/jira/rest/api/2/issue/12461400","fields":{"summary":"Testcase failures in TextException on MSSQL","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/6","id":"6","description":"A new unit, integration or system test.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/requirement.png","name":"Test","subtask":false}}}},{"id":"12330743","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12330743","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12417320","key":"OPENJPA-991","self":"https://issues.apache.org/jira/rest/api/2/issue/12417320","fields":{"summary":"Implement LockTimeout using db specific server-side lock timeout feature","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/4","description":"This issue was once resolved, but the resolution was deemed incorrect. From here issues are either marked assigned or resolved.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/reopened.png","name":"Reopened","id":"4","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/2","id":2,"key":"new","colorName":"blue-gray","name":"To Do"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/4","id":"4","description":"An improvement or enhancement to an existing feature or task.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21140&avatarType=issuetype","name":"Improvement","subtask":false,"avatarId":21140}}}}],"customfield_12313825":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"customfield_12312337":null,"customfield_12313823":null,"customfield_12312338":null,"customfield_12311920":null,"customfield_12313822":null,"customfield_12312335":null,"customfield_12313821":null,"customfield_12312336":null,"customfield_12313820":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/12311300","id":"12311300","name":"jdbc","description":"jdbc interface"},{"self":"https://issues.apache.org/jira/rest/api/2/component/12311304","id":"12311304","name":"jpa","description":"Java Persistence API interface"},{"self":"https://issues.apache.org/jira/rest/api/2/component/12311302","id":"12311302","name":"kernel","description":"Kernel"},{"self":"https://issues.apache.org/jira/rest/api/2/component/12311309","id":"12311309","name":"query","description":"Issues related to query execution, including the JPQL parser and internal query representation."},{"self":"https://issues.apache.org/jira/rest/api/2/component/12311310","id":"12311310","name":"sql","description":"Issues to do with SQL, including generation of SQL. Issues that have to do with JDBC (i.e., execution of SQL statements and result set processing) should be filed in the jdbc component."}],"customfield_12312026":null,"customfield_12312023":null,"customfield_12312024":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12310921":null,"customfield_12310920":"203550","customfield_12312823":null,"creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"subtasks":[{"id":"12459907","key":"OPENJPA-1594","self":"https://issues.apache.org/jira/rest/api/2/issue/12459907","fields":{"summary":"Tests not handling new QueryTimeOut and LockTimeOut exceptions correctly","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/7","id":"7","description":"The sub-task of the issue","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21146&avatarType=issuetype","name":"Sub-task","subtask":true,"avatarId":21146}}}],"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"aggregateprogress":{"progress":0,"total":0},"customfield_12313520":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"customfield_12313924":null,"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/OPENJPA-1565/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12313920":null,"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133},"timespent":null,"customfield_12314020":"{summaryBean=com.atlassian.jira.plugin.devstatus.rest.SummaryBean@2b293ed6[summary={pullrequest=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@2741de15[overall=PullRequestOverallBean{stateCount=0, state='OPEN', details=PullRequestOverallDetails{openCount=0, mergedCount=0, declinedCount=0}},byInstanceType={}], build=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@1e7680dc[overall=com.atlassian.jira.plugin.devstatus.summary.beans.BuildOverallBean@37ee5471[failedBuildCount=0,successfulBuildCount=0,unknownBuildCount=0,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], review=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@774c530b[overall=com.atlassian.jira.plugin.devstatus.summary.beans.ReviewsOverallBean@5555c0d9[stateCount=0,state=<null>,dueDate=<null>,overDue=false,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], deployment-environment=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@7f0fcacf[overall=com.atlassian.jira.plugin.devstatus.summary.beans.DeploymentOverallBean@3a874483[topEnvironments=[],showProjects=false,successfulCount=0,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], repository=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@2f469cd3[overall=com.atlassian.jira.plugin.devstatus.summary.beans.CommitOverallBean@546ec2ea[count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], branch=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@55b6231a[overall=com.atlassian.jira.plugin.devstatus.summary.beans.BranchOverallBean@5515c673[count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}]},errors=[],configErrors=[]], devSummaryJson={\"cachedValue\":{\"errors\":[],\"configErrors\":[],\"summary\":{\"pullrequest\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"stateCount\":0,\"state\":\"OPEN\",\"details\":{\"openCount\":0,\"mergedCount\":0,\"declinedCount\":0,\"total\":0},\"open\":true},\"byInstanceType\":{}},\"build\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"failedBuildCount\":0,\"successfulBuildCount\":0,\"unknownBuildCount\":0},\"byInstanceType\":{}},\"review\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"stateCount\":0,\"state\":null,\"dueDate\":null,\"overDue\":false,\"completed\":false},\"byInstanceType\":{}},\"deployment-environment\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"topEnvironments\":[],\"showProjects\":false,\"successfulCount\":0},\"byInstanceType\":{}},\"repository\":{\"overall\":{\"count\":0,\"lastUpdated\":null},\"byInstanceType\":{}},\"branch\":{\"overall\":{\"count\":0,\"lastUpdated\":null},\"byInstanceType\":{}}}},\"isStale\":false}}","customfield_12314141":null,"customfield_12314140":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/12310351","id":"12310351","key":"OPENJPA","name":"OpenJPA","projectTypeKey":"software","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=12310351&avatarId=10043","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310351&avatarId=10043","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310351&avatarId=10043","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310351&avatarId=10043"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10252","id":"10252","description":"","name":"OpenJPA"}},"aggregatetimespent":null,"customfield_12312520":null,"customfield_12314422":null,"customfield_12314421":null,"customfield_12314146":null,"customfield_12314420":null,"customfield_12314145":null,"customfield_12314144":null,"customfield_12314143":null,"resolutiondate":"2010-03-22T20:11:05.161+0000","workratio":-1,"customfield_12312923":null,"customfield_12312920":null,"customfield_12312921":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/OPENJPA-1565/watchers","watchCount":0,"isWatching":false},"created":"2010-03-11T22:51:41.458+0000","updated":"2010-04-07T15:18:17.684+0000","timeoriginalestimate":null,"description":"Narrowing SQL Exception to a more specific exception such as lock or query or referential integrity violation does not distinguish correctly whether a query or lock request has timed out. \nThis distinction is critical for JPA 2.0 spec compliance because QueryTimeOut and LockTimeOut exceptions are not supposed to cause rollback as per the spe (Section 3.9).\n\n","customfield_10010":null,"timetracking":{},"customfield_12314523":null,"customfield_12314127":null,"customfield_12314522":null,"customfield_12314126":null,"customfield_12314521":null,"customfield_12314125":null,"customfield_12314520":null,"customfield_12314124":null,"attachment":[],"customfield_12312340":null,"customfield_12314123":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12314122":null,"customfield_12314121":null,"customfield_12310041":null,"customfield_12314120":null,"customfield_12314129":null,"customfield_12314524":null,"customfield_12314128":null,"summary":"QueryTimeOut and LockTimeOut exceptions are not raised correctly","customfield_12314130":null,"customfield_12310291":null,"customfield_12310290":null,"customfield_12314138":null,"customfield_12314137":null,"environment":null,"customfield_12314136":null,"customfield_12314135":null,"customfield_12311020":null,"customfield_12314134":null,"duedate":null,"customfield_12314132":null,"customfield_12314131":null,"comment":{"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12458864/comment/12845979","id":"12845979","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"body":"Looks like this is causing new junit test failures on any DB besides Derby.\n\nMySQL 2 tests -\n1) TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError was raised but expected javax.persistence.QueryTimeoutException\nat org.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat org.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities     (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError was raised but expected javax.persistence.LockTimeoutException\nat org.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat org.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\nMicrosoft SQL 2008 - 1 test -\nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities     (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError was raised but expected javax.persistence.LockTimeoutException\nat org.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat org.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\n\nDB2 - 2 tests -\n1) TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError was raised but expected javax.persistence.QueryTimeoutException\nat org.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat org.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities     (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError was raised but expected javax.persistence.LockTimeoutException\nat org.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat org.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"created":"2010-03-16T16:32:47.856+0000","updated":"2010-03-16T16:35:11.728+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12458864/comment/12845984","id":"12845984","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"1. Note the SQL Error State and Code of the SQL Exception when it fails to throw expected exception\n2. Check sql-error-state-codes.xml for that database dictionary. The error states are categorized in that file. If the observed error does not occur in the list of error states, add it (the error state not the error code).\n3. Also check corresponding DB dictionary implementation isFatalException() and ensure that it returns false when query/lock timeouts are intended to be raised. \n4. Returning false from isFatalException() will ensure that at facade level the transaction is not rolled back.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2010-03-16T16:44:04.196+0000","updated":"2010-03-16T16:44:04.196+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12458864/comment/12846005","id":"12846005","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"I ran with mysql-connector-java-5.1.6 locally and the tests passed for \nMySQL.\n\nI had given Jody a patch for DB2 errors -- will check if he can verify\n\nCan you please send me the stacktrace of SQL error for  MS-SQL 2008?\n\nRegards --\n\nPinaki \n\n\n\n\n\n\n\n\"Donald Woods (JIRA)\" <jira@apache.org> \n03/16/2010 11:36 AM\n\nTo\nPinaki Poddar/Dallas/IBM@IBMUS\ncc\n\nSubject\n[jira] Issue Comment Edited: (OPENJPA-1565) QueryTimeOut and LockTimeOut \nexceptions are not raised correctly\n\n\n\n\n\n\n\n    [ \nhttps://issues.apache.org/jira/browse/OPENJPA-1565?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel&focusedCommentId=12845979#action_12845979 \n] \n\nDonald Woods edited comment on OPENJPA-1565 at 3/16/10 4:35 PM:\n----------------------------------------------------------------\n\nLooks like this is causing new junit test failures on any DB besides \nDerby.\n\nMySQL 2 tests -\n1) \nTestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities \n   (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.QueryTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) \nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities \n    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.LockTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\nMicrosoft SQL 2008 - 1 test -\nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities \n    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.LockTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\n\nDB2 - 2 tests -\n1) \nTestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities \n   (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.QueryTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) \nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities \n    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.LockTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\n\n      was (Author: drwoods):\n    Looks like this is causing new junit test failures on any DB besides \nDerby.\n\nMySQL 2 tests -\n1) \nTestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities \n   (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.QueryTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) \nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities \n    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.LockTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\nMicrosoft SQL 2008 - 1 test -\nTestException.testThrowsEntityExistsException \n(org.apache.openjpa.persistence.exception)\njunit.framework.AssertionFailedError: <openjpa-2.0.0-SNAPSHOT-runknown \nfatal store error> org.apache.openjpa.persistence.RollbackException: The \ntransaction has been rolled back. See the nested exceptions for details on \nthe errors that occurred. or its cause is not instanceof class \njavax.persistence.EntityExistsException\nat junit.framework.Assert.fail(Assert.java:47)\nat \norg.apache.openjpa.persistence.exception.TestException.assertException(TestException.java:207)\n\nDB2 - 2 tests -\n1) \nTestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities \n   (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.QueryTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testQueryTimeOutExceptionWhileQueryingWithLocksOnAlreadyLockedEntities(TestTimeoutException.java:63)\n2) \nTestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities \n    (org.apache.openjpa.persistence.query)\njunit.framework.AssertionFailedError: junit.framework.AssertionFailedError \nwas raised but expected javax.persistence.LockTimeoutException\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.assertError(TestTimeoutException.java:141)\nat \norg.apache.openjpa.persistence.query.TestTimeoutException.testLockTimeOutExceptionWhileLockingAlreadyLockedEntities(TestTimeoutException.java:93)\n\n \n2.0.0-beta2\nquery or referential integrity violation does not distinguish correctly \nwhether a query or lock request has timed out. \nQueryTimeOut and LockTimeOut exceptions are not supposed to cause rollback \nas per the spe (Section 3.9).\n\n-- \nThis message is automatically generated by JIRA.\n-\nYou can reply to this email to add a comment to the issue online.\n\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2010-03-16T17:14:29.937+0000","updated":"2010-03-16T17:14:29.937+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12458864/comment/12848312","id":"12848312","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"body":"Code committed by Pinaki.  Remaining test issues will be handled via OPENJPA-1594.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"created":"2010-03-22T20:11:05.156+0000","updated":"2010-03-22T20:11:05.156+0000"}],"maxResults":4,"total":4,"startAt":0},"customfield_12311820":"0|i0z7pb:","customfield_12314139":null}}

